src/service/url-replacements-impl.js

```diff
@@ -584,6 +590,21 @@ export class GlobalVariableSource extends VariableSource {
     });
   }
 
+  /**
+   * Merges any replacement parameters into a given URL's query string,
+   * preferring values set in the original query string.
+   * @param {string} orig The original URL
+   * @return {string} The resulting URL
+   */
+  mergeReplaceParamsIntoUrl_(orig) {
+    const {extraParams} =
+        /** @type {!Object} */ (Services.documentInfoForDoc(this.ampdoc));
```

Comment:
Hmm, weird that you'd need this type cast.

Reply:
I actually battled this one for a little while. There's a lot of time pressure on this PR, so I figure it's better to get this out now. But I'm more than happy to clean this up in a follow up (maybe we can chat offline about the type situation?)